source/includes/fundamentals/code-snippets/bson-extra-elements/ProductWithBsonExtraElements.java

```diff
@@ -0,0 +1,75 @@
+package fundamentals;
+
+import java.util.List;
+
+import org.bson.BsonType;
+import org.bson.Document;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonDiscriminator;
+import org.bson.codecs.pojo.annotations.BsonExtraElements;
+import org.bson.codecs.pojo.annotations.BsonId;
+import org.bson.codecs.pojo.annotations.BsonIgnore;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+import org.bson.codecs.pojo.annotations.BsonRepresentation;
+
+@BsonDiscriminator(value="ProductWithBsonExtraElements", key="_cls")
+public class ProductWithBsonExtraElements {
+
+    @BsonProperty("modelName")
+    private String name;
+
+    @BsonId()
+    @BsonRepresentation(BsonType.OBJECT_ID)
+    private String serialNumber;
+
+    @BsonIgnore
```

Comment:
Does this property serve any purpose in the example?  If not, it should be removed.

Reply:
The context of the example mentions an addition to the ProductPojo, so keeping everything else was implied. It's not strictly necessary, but would likely make more sense to the next writer.